### PR TITLE
[c10d] C10d release to torch.distributed for PT1

### PR DESCRIPTION
**Pull request by teng-li**
The old `torch.distributed` will go to `torch.distributed.deprecated`
The old DDP will go to `torch.nn.parallel.deprecated`

Now `torch.nn.parallel.DDP` will use c10d DDP
Now `torch.distributed` will use C10d frontend API

### CODE REVIEW

**Comment by teng-li** [comment]
@pietern This PR should be good to go. Please review/stamp

**Comment by myleott** [comment]
Please make sure to merge the gradient normalization change from here: https://github.com/pytorch/pytorch/pull/11109

**Review by pietern — APPROVED** [review]
